src/components/text-editor/text-editor/text-editor.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    State,
+    h,
+} from '@stencil/core';
+import { EditorState } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { Schema, DOMParser } from 'prosemirror-model';
+import { schema } from 'prosemirror-schema-basic';
+import { exampleSetup } from 'prosemirror-example-setup';
+
+/**
+ * The `limel-text-editor` component wraps the toast-ui react editor. This editor
+ * allows a rich text editing experience with markdown support.
+ *
+ * @exampleComponent limel-example-basic-text-editor
+ * @beta
+ */
+@Component({
+    tag: 'limel-text-editor',
+    shadow: true,
+    styleUrl: 'text-editor.scss',
+})
+export class TextEditor {
```

Comment:
Maybe we should implement `FormComponent` here, as they potentially wanted the editor in input fields 🤔

Reply:
Yeah, that sounds like a good idea.

(Some components have a wrapper in /src/form/widgets/ instead, but I think that's probably only components that we already had before we added limel-form.)

Reply:
Did this in a separate PR, was quite a large change